src/transformers/models/patchtst/modeling_patchtst.py

```diff
@@ -0,0 +1,2103 @@
+            module.bias.data.zero_()
+            module.weight.data.fill_(1.0)
+        elif isinstance(module, (nn.Linear, nn.Conv1d)):
+            module.weight.data.normal_(mean=0.0, std=self.config.init_std)
+            if module.bias is not None:
+                module.bias.data.zero_()
+
+    def _set_gradient_checkpointing(self, module, value=False):
+        if isinstance(module, (PatchTSTEncoder)):
+            module.gradient_checkpointing = value
+
+
+class PatchTSTEncoder(PatchTSTPreTrainedModel):
+    """
+    PatchTST Encoder
+    """
+
+    def __init__(self, config: PatchTSTConfig):
+        super().__init__(config)
+        self.num_input_channels = config.num_input_channels
+        self.num_patches = config.num_patches
+        self.patch_length = config.patch_length
+        self.d_model = config.d_model
+        self.shared_embedding = config.shared_embedding
+        self.use_cls_token = config.use_cls_token
+        self.gradient_checkpointing = False
+
+        # Input encoding: projection of feature vectors onto a d-dim vector space
+        if not config.shared_embedding:
+            self.input_embedding = nn.ModuleList()
+            for _ in range(self.num_input_channels):
+                self.input_embedding.append(nn.Linear(config.patch_length, config.d_model))
+        else:
+            self.input_embedding = nn.Linear(config.patch_length, config.d_model)
+
+        # Positional encoding
+        if config.use_cls_token:
+            self.cls_token = nn.Parameter(torch.zeros(1, 1, 1, config.d_model))
+            self.position_enc = positional_encoding(
+                config.positional_encoding, config.learn_pe, config.num_patches + 1, config.d_model
+            )
+        else:
+            self.position_enc = positional_encoding(
+                config.positional_encoding, config.learn_pe, config.num_patches, config.d_model
+            )
+
+        # Positional dropout
+        self.positional_dropout = (
+            nn.Dropout(config.positional_dropout) if config.positional_dropout > 0 else nn.Identity()
+        )
+
+        # Encoder
+        self.encoder = PatchTSTEncoderBlock(config)
+
+        # Initialize weights and apply final processing
+        self.post_init()
+
+    def forward(
+        self, past_values: torch.Tensor, output_hidden_states: Optional[bool] = None
+    ) -> BaseModelOutputWithNoAttention:
+        """
+        Parameters:
+            past_values (`torch.Tensor` of shape `(batch_size, num_channels, num_patches, patch_length)`, *required*):
+                Past values of the time series
+            output_hidden_states (bool, optional): Indicates if hidden states should be output.
+
+        return:
+            `BaseModelOutputWithNoAttention`
+        """
+        _, num_input_channels, _, _ = past_values.shape
+
+        output_hidden_states = (
+            output_hidden_states if output_hidden_states is not None else self.config.output_hidden_states
+        )
```

Comment:
at this point `output_hidden_states` is either True or False

Reply:
This wasn't resolved